spec.bs

```diff
@@ -201,68 +203,247 @@ This document introduces a new storage API that is intentionally not partitioned
 
 The {{SharedStorageWorklet}} Interface {#worklet}
 =================================================
-The {{SharedStorageWorklet}} object allows developers to supply [=module scripts=] to process [=Shared Storage=] data and then output the result through one or more of the output gates. Currently there are two output gates, the [=private aggregation=] output gate and the {{WindowSharedStorage/selectURL()|URL-selection}} output gate.
+The {{SharedStorageWorklet}} object allows developers to supply [=module scripts=] to process [=Shared Storage=] data and then output the result through one or more of the output gates. Currently there are two output gates, the [=private aggregation=] output gate and the {{SharedStorageWorklet/selectURL()|URL-selection}} output gate.
+
+<xmp class='idl'>
+  typedef (USVString or FencedFrameConfig) SharedStorageResponse;
+</xmp>
 
 <xmp class='idl'>
   [Exposed=(Window)]
   interface SharedStorageWorklet : Worklet {
+    Promise<SharedStorageResponse> selectURL(DOMString name,
+                                 FrozenArray<SharedStorageUrlWithMetadata> urls,
+                                 optional SharedStorageRunOperationMethodOptions options = {});
+    Promise<any> run(DOMString name,
+                     optional SharedStorageRunOperationMethodOptions options = {});
   };
 </xmp>
 
-Each {{SharedStorageWorklet}} has an associated boolean <dfn>addModule initiated</dfn>, initialized to false.
+Each {{SharedStorageWorklet}} has an associated boolean <dfn for="SharedStorageWorklet">addModule initiated</dfn>, initialized to false.
+
+Each {{SharedStorageWorklet}} has an associated [=/origin=] <dfn for="SharedStorageWorklet">worklet origin</dfn>, initialized to null. It will be set to the |moduleURL|'s [=url/origin=] when {{Worklet/addModule()|addModule}}(|moduleURL|, <var ignore=''>options</var>) is called.
```

Comment:
```suggestion
Each {{SharedStorageWorklet}} has an associated [=/origin=]-or-null <dfn for="SharedStorageWorklet">worklet origin</dfn>, initialized to null.

Note: It will be set to the worklet script's [=URL/origin=] during {{Worklet/addModule()}}.
```